pkg/rid/store/cockroach/store.go

```diff
@@ -158,5 +162,8 @@ func (s *Store) CleanUp(ctx context.Context) error {
 // GetVersion returns the Version string for the Database.
 // If the DB was is not bootstrapped using the schema manager we throw and error
 func (s *Store) GetVersion(ctx context.Context) (*semver.Version, error) {
+	if s.version != nil {
```

Comment:
I think in light of the issue we see with transactions being aborted due to bubbling up the error too quickly with stack tracing, it might be a good idea to revisit this. as for what happens to the DSS instance when the db is migrated from under it? I think the schema will reject the call if there is a major change to it. Either case the schema should only be migrated after the supported code has been rolled out where the code should be backwards compatible to the older schema. But I can also see that we might not get that all the time in the real world.

Reply:
What I'm thinking is that if we cache the DB version, then the 3.0.0 interaction methods will be used even after the DB gets migrated to 3.1.0.  So, someone may be running the newest code, but all their endpoints will start failing as soon as the DB is migrated from 3.0.0 to 3.1.0 since their endpoints won't recognize the DB version change, right?

Reply:
If we have database v3.1.0 run with the old code, the system will not fail. The system just doesn't update 'writer' field.

Reply:
Thats true since we are not enforcing that the writer column must not be null the call should be successful with the column written to be null. when we do move to V4.0.0 which we plan to enforce the column then at that point the code will fail. it is after all the next major version

Reply:
This seems complex enough to warrant a note in build/README.md -- I think we should explain what will work (new code with 3.0.0, new code with 3.1.0), what won't work (old code with 3.1.0), and what works with what caveats (new code with container started when DB was 3.0.0 then updated to 3.1.0; caveat that `writer` is omitted).

Reply:
Update readme file